spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonTracingAutoConfiguration.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.tracing.NoOpSpanFactory;
+import org.axonframework.tracing.SpanAttributesProvider;
+import org.axonframework.tracing.SpanFactory;
+import org.axonframework.tracing.attributes.AggregateIdentifierSpanAttributesProvider;
+import org.axonframework.tracing.attributes.MessageIdSpanAttributesProvider;
+import org.axonframework.tracing.attributes.MessageNameSpanAttributesProvider;
+import org.axonframework.tracing.attributes.MessageTypeSpanAttributesProvider;
+import org.axonframework.tracing.attributes.MetadataSpanAttributesProvider;
+import org.axonframework.tracing.attributes.PayloadTypeSpanAttributesProvider;
+import org.axonframework.tracing.opentelemetry.OpenTelemetrySpanFactory;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Configures tracing for Axon Framework. Defaults to the {@link NoOpSpanFactory}. If the
+ * {@link OpenTelemetrySpanFactory} is on the classpath, uses Open Telemetry instead.
+ * <p>
+ * For Open tracing, take a look at the <a href="https://github.com/AxonFramework/extension-tracing">Open Tracing
+ * extension</a>
+ */
+@Configuration
+@AutoConfigureBefore({AxonServerAutoConfiguration.class, AxonAutoConfiguration.class})
+public class AxonTracingAutoConfiguration {
```

Comment:
I understand it constructs an additional configuration class, but for this to be in line with the `MessageMonitor` auto-configuration I'd expect an `OpenTelemetryAutoConfiguration` for the `OpenTelemetrySpanFactory`.

Furthermore, I think the base `SpanFactory` and `SpanAttributesProviders` might be better placed in the base auto-config class `AxonAutoConfiguration`, because they're core to a lot of components through this change.

Reply:
I agree that the autoconfiguration classes should be split, and I have done that. But I would like to keep them split from the `AxonAutoConfiguration`, because it's quite a few beans due to the `SpanAttributesProviders`

Reply:
Fair enough, that makes sense. Thanks for splitting it up regardless :+1: